schema/pet-insurance.json

```diff
@@ -328,90 +1037,134 @@
+                },
+                "selectedPaymentDay": {
+                    "type": "string",
+                    "description": "Preferred payment day, one of availableCardPaymentDays.",
+                    "example": "10"
                 }
             },
             "required": [
-                "text",
-                "price"
-            ]
+                "card",
+                "person",
+                "address"
+            ],
+            "additionalProperties": false
         },
-        "VetFeesLimit": {
-            "type": "array",
-            "description": "",
+        "DirectDebitPayment": {
+            "type": "object",
+            "description": "Account details for direct debit.",
             "pii": true,
-            "minItems": 1,
-            "items": {
-                "$ref": "#/domains/PetInsurance/types/VetFeeLimitItem"
-            }
+            "properties": {
+                "sortCode": {
+                    "type": "string",
+                    "description": "Bank sort code",
+                    "pattern": "\\d{2}-?\\d{2}-?\\d{2}",
+                    "example": "56-00-29"
+                },
+                "accountNumber": {
+                    "type": "string",
+                    "minLength": 8,
+                    "maxLength": 8,
+                    "description": "Cover marketing name as provided by the insurer.",
+                    "example": "26207729"
+                },
+                "accountHolder": {
+                    "$ref": "#/domains/Generic/types/Person"
+                },
+                "accountHolderAddress": {
+                    "$ref": "#/domains/Generic/types/Address"
+                },
+                "selectedPaymentDay": {
+                    "type": "string",
+                    "description": "Preferred payment day, one of availableDirectDebitPaymentDays.",
+                    "example": "10"
+                }
+            },
+            "required": [
+                "sortCode",
+                "accountNumber",
+                "accountHolder",
+                "accountHolderAddress"
+            ],
+            "additionalProperties": false
         },
-        "VetFeeLimitItem": {
+        "VetFee": {
             "type": "object",
-            "description": "",
             "additionalProperties": false,
+            "description": "",
             "properties": {
                 "text": {
                     "type": "string",
                     "description": "",
-                    "example": "Classic"
+                    "example": "Accident Only"
                 },
                 "price": {
-                    "typeRef": "#/domains/Generic/types/Price"
+                    "$ref": "#/domains/Generic/types/Price"
                 }
             },
             "required": [
                 "text",
                 "price"
             ]
         },
+        "AvailablePaymentDays": {

```

Comment:
Can payment days be missing here? If not, remove `null`.

Reply:
No. How do I implement? With required?

Reply:
Make `AvailablePaymentDays` an object `{"type": "array", "items": ...}`